Gem/Nodes/Scenes/ImportNodeScene.swift

```diff
@@ -0,0 +1,171 @@
+// Copyright (c). Gem Wallet. All rights reserved.
+
+import SwiftUI
+import Components
+import Style
+
+struct ImportNodeScene: View {
+    @Environment(\.dismiss) private var dismiss
+
+    @StateObject var model: ImportNodeSceneViewModel
+    @State private var isPresentingScanner: Bool = false
+    @State private var isPresentingErrorAlert: String?
+
+    @FocusState private var focusedField: Field?
+
+    var onDismiss: (() -> Void)
+
+    enum Field: Int, Hashable {
+        case address
+    }
+
+    var body: some View {
+        VStack {
+            List {
+                inputSection
+                nodeInfoView
+            }
+            Spacer()
+            StatefullButton(
+                text: Localized.Wallet.Import.action,
+                viewState: model.state,
+                action: onSelectImport
+            )
+            .disabled(model.shouldDisableImportButton)
+            .frame(maxWidth: Spacing.scene.button.maxWidth)
+        }
+        .padding(.bottom, Spacing.scene.bottom)
+        .background(Colors.grayBackground)
+        .frame(maxWidth: .infinity)
+        .navigationTitle(model.title)
+        .navigationBarTitleDisplayMode(.inline)
+        .toolbar {
+            ToolbarItem(placement: .navigationBarLeading) {
+                Button(Localized.Common.done, action: onSelectDone)
+                    .bold()
+            }
+            ToolbarItem(placement: .navigationBarTrailing) {
+                Button(action: onSelectScan) {
+                    Image(systemName: SystemImage.qrCode)
+                }
+                .bold()
+            }
+        }
+        .sheet(isPresented: $isPresentingScanner) {
+            ScanQRCodeNavigationStack(isPresenting: $isPresentingScanner, action: onScanFinished(_:))
+        }
+        .alert(item: $isPresentingErrorAlert) {
+            Alert(title: Text(""), message: Text($0))
+        }
+    }
+}
+
+// MARK: - UI Components
+
+extension ImportNodeScene {
+    private var inputSection: some View {
```

Comment:
can you create an issue to create this custom view? it's used in many places in the app, need to have reusability, recipient, custom address, here

Reply:
https://github.com/orgs/gemwalletcom/projects/2/views/4?pane=issue&itemId=67986270